### DISCUSSION TITLE
Activating third block not working

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#28740] [Q&A Modules: Phase field]
Hello, I am trying to activate three blocks, each of them after a certain amount of time in the simulation. For the first 2 blocks everything is working and the heat conduction and heat source are correct, but when I try to activate the third block (after 640 seconds) the block does not activate its properties. This is part of the input file:
```
[Kernels]
  [./time]
    type = ADHeatConductionTimeDerivative
    variable = T
    block = '1'
  [../]
  [./heat_conduct]
    type = ADHeatConduction
    variable = T
    thermal_conductivity = thermal_conductivity
    block = '1'
  [../]
  [./heat_source]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
    block = '1'
  [../]
  [./time2]
    type = ADHeatConductionTimeDerivative
    variable = T
    block = '2'
  [../]
  [./heat_conduct2]
    type = ADHeatConduction
    variable = T
    thermal_conductivity = thermal_conductivity
    block = '2'
  [../]
  [./heat_source2]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
    block = '2'
  [../]
  [./time3]
    type = ADHeatConductionTimeDerivative
    variable = T
    block = '3'
  [../]
  [./heat_conduct3]
    type = ADHeatConduction
    variable = T
    thermal_conductivity = thermal_conductivity
    block = '3'
  [../]
  [./heat_source3]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
    block = '3'
  [../]
[]

[Functions]
  [./conditional_function]
    type = ParsedFunction
    expression = 't <= 120'
  [../]
  [./conditional_function2]
    type = ParsedFunction
    expression = 'if(t > 120,if(t <= 640,1,0),0)'
  [../]
  [./conditional_function3]
    type = ParsedFunction
    expression = 't > 640'
  [../]
[]

[Controls]
  [./u_threshold]
    type = ConditionalFunctionEnableControl
    conditional_function = conditional_function
    disable_objects = 'Kernels::time2 Kernels::heat_conduct2 Kernels::heat_source2 Kernels::time3 Kernels::heat_conduct3 Kernels::heat_source3'
    enable_objects = 'Kernels::time Kernels::heat_conduct Kernels::heat_source'
  [../]
  [./u_threshold2]
    type = ConditionalFunctionEnableControl
    conditional_function = conditional_function2
    disable_objects = 'Kernels::time3 Kernels::heat_conduct3 Kernels::heat_source3'
    enable_objects = 'Kernels::time2 Kernels::heat_conduct2 Kernels::heat_source2'
  [../]
  [./u_threshold3]
    type = ConditionalFunctionEnableControl
    conditional_function = conditional_function3
    enable_objects = 'Kernels::time3 Kernels::heat_conduct3 Kernels::heat_source3'
  [../]
[]
```

**Comment by GiudGiud** [comment]
I wonder if the u_threshold_2 is executed after the _3 and disabling the kernels::xyz3 objects.
Let's try to chain them.
Add the `depends_on` parameter. 3 depends on 2, which depends on 1

**Reply by fervaldies** [reply]
Thank you for your answer. I have tried this, but no the first two conditions are not acting at all and it is only after 640s that the third one is activated:

[Kernels]
  [./time]
    type = ADHeatConductionTimeDerivative
    variable = T
    block = '1'
  [../]
  [./heat_conduct]
    type = ADHeatConduction
    variable = T
    thermal_conductivity = thermal_conductivity
    block = '1'
  [../]
  [./heat_source]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
    block = '1'
  [../]
  [./time2]
    type = ADHeatConductionTimeDerivative
    variable = T
    block = '2'
  [../]
  [./heat_conduct2]
    type = ADHeatConduction
    variable = T
    thermal_conductivity = thermal_conductivity
    block = '2'
  [../]
  [./heat_source2]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
    block = '2'
  [../]
  [./time3]
    type = ADHeatConductionTimeDerivative
    variable = T
    block = '3'
  [../]
  [./heat_conduct3]
    type = ADHeatConduction
    variable = T
    thermal_conductivity = thermal_conductivity
    block = '3'
  [../]
  [./heat_source3]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
    block = '3'
  [../]
[]

[Functions]
  [./conditional_function]
    type = ParsedFunction
    expression = 't <= 120'
  [../]
  [./conditional_function2]
    type = ParsedFunction
    expression = 'if(t > 120,if(t <= 640,1,0),0)'
  [../]
  [./conditional_function3]
    type = ParsedFunction
    expression = 't > 640'
  [../]
[]

[Controls]
  [./u_threshold]
    type = ConditionalFunctionEnableControl
    conditional_function = conditional_function
    disable_objects = 'Kernels::time2 Kernels::heat_conduct2 Kernels::heat_source2 Kernels::time3 Kernels::heat_conduct3 Kernels::heat_source3'
    enable_objects = 'Kernels::time Kernels::heat_conduct Kernels::heat_source'
  [../]
  [./u_threshold2]
    type = ConditionalFunctionEnableControl
    conditional_function = conditional_function2
    disable_objects = 'Kernels::time3 Kernels::heat_conduct3 Kernels::heat_source3'
    enable_objects = 'Kernels::time Kernels::heat_conduct Kernels::heat_source Kernels::time2 Kernels::heat_conduct2 Kernels::heat_source2'
    depends_on = u_threshold
  [../]
  [./u_threshold3]
    type = ConditionalFunctionEnableControl
    conditional_function = conditional_function3
    #disable_objects = 'Kernels::time Kernels::heat_conduct Kernels::heat_source Kernels::time2 Kernels::heat_conduct2 Kernels::heat_source2'
    enable_objects = 'Kernels::time Kernels::heat_conduct Kernels::heat_source Kernels::time2 Kernels::heat_conduct2 Kernels::heat_source2 Kernels::time3 Kernels::heat_conduct3 Kernels::heat_source3'
    depends_on = u_threshold2
  [../]
[]

**Comment by fervaldies** [comment]
Okay I think it is working now. I just disabled the previous conditions and created new kernels that are activating the 2 or 3 blocks for each time, rather than having different kernels that only activate one block each. Thanks!
